docs/RFCS/structured_configs.md

```diff
@@ -0,0 +1,141 @@
+- Feature Name: Configs for structured data
+- Status: draft
+- Start Date: 2015-08-19
+- RFC PR: [#2183](https://github.com/cockroachdb/cockroach/pull/2183)
+- Cockroach Issue: [#2090](https://github.com/cockroachdb/cockroach/issues/2090)
+
+# Summary
+
+This RFC describes the application of various configurations to the structured data
+representation. We address storage, distribution, and modification.
+
+# Motivation
+
+Structured data will soon be the only publicly accessible API. This requires a change
+in the way configurations are specified and applied, moving from prefix-based logic
+to databases and tables.
+
+The following configs currently exist:
+* Accounting: usage limits (bytes, counts). Not currently used.
+* Permissions: user read/write permissions. Used by the KV endpoint.
+* Users: password storage for the web interface. Currently settable, but not used.
+* Zones: replication configuration. Used to make allocation decisions.
+
+# Prerequisites
+
+Complete implementation of this RFC depends on the following:
+* Gossip system database: [#2179](https://github.com/cockroachdb/cockroach/issues/2179)
+* Disable KV endpoint: [#2089](https://github.com/cockroachdb/cockroach/issues/2089)
+
+# Detailed design
+
+## Storage
+
+We propose moving all configuration files to the `system` database as separate tables.
+
+The table schema depends on the configuration. Some (eg: `users`) apply globally,
+while some (eg: `accounting`, `zones`) apply to databases and tables.
+
+For each configuration, we propose the following schemas:
+
+#### Accounting
+
+The accounting table applies to databases and potentially to tables as well.
+Its contents have yet to be determined.
```

Comment:
Ah, missed that part of the motivation. Perhaps "quotas" would be a more descriptive name than accounting.